ast-checks/src/comment_finders.rs

```diff
@@ -0,0 +1,57 @@
+use rnix::{NodeOrToken, SyntaxKind::*, SyntaxNode};
+
+use crate::tree_utils::{next_siblings, prev_siblings, walk_kind};
+
+
+pub fn find_comment_within(node: &SyntaxNode) -> Option<String> {
+    let mut maybe_comment_token = None;
+    for item in walk_kind(node, TOKEN_COMMENT) {
+        match item {
+            NodeOrToken::Token(token) => {
+                maybe_comment_token = Some(token);
+                break;
+            }
+            NodeOrToken::Node(_) => {
+                unreachable!("");
+            }
+        }
+    }
+    let tok = maybe_comment_token?;
+    let mut comments = vec![tok.text().clone()];
+
+    for elem in next_siblings(&NodeOrToken::Token(tok)) {
+        match elem {
+            NodeOrToken::Token(tok) if tok.kind() == TOKEN_COMMENT => {
+                comments.push(tok.text().clone());
+            }
+            NodeOrToken::Token(tok) if tok.kind().is_trivia() => (),
+            _ => break,
+        }
+    }
+
+    let doc = comments
+        .iter()
+        .map(|it| it.trim_start_matches('#').trim())
+        .collect::<Vec<_>>()
+        .join("\n");
+    return Some(doc).filter(|it| !it.is_empty());
```

Comment:
You don't need return statements. Just write `Some(...)` (yes, no semicolon in the end)